src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -665,11 +665,18 @@ private function getAdjacentRecordID($offset)
      */
     private function getNumPages(GridField $gridField): int
     {
-        return $gridField
-                ->getConfig()
-                ->getComponentByType(GridFieldPaginator::class)
-                ->getTemplateParameters($gridField)
-                ->toMap()['NumPages'];
+        /** @var GridFieldPaginator $component */
+        $component = $gridField
+            ->getConfig()
+            ->getComponentByType(GridFieldPaginator::class);
+        if (is_null($component)) {
+            return 1;
+        }
+        $params = $component->getTemplateParameters($gridField);
+        if (is_null($params)) {
+            return 1;
+        }
+        return $params->toMap()['NumPages'];
```

Comment:
I appreciate the better legibility of the convoluted approach, but would this work as well?
```
    return $gridField
            ->getConfig()
            ->getComponentByType(GridFieldPaginator::class)
            ->getTemplateParameters($gridField)
            ->toMap()['NumPages'] ?? 1;
```

Reply:
It's spread out like this because I'm also fixing the case where there's not a GridFieldPaginator in the GridField config for whatever reason.  Also lets us use `@var GridFieldPaginator $component` for IDE type hinting.

Reply:
If that works, I'd prefer it as well.

Reply:
I'd have loved the null coallescing example to work but alas it doesn't catch null up the chain.
I'm okay with what's there currently and since this is blocking you right now Steve I'll merge this as is.

@michalkleiner if you do feel strongly about changing this feel free to comment, I'll be happy to raise a follow-up PR which changes the formatting of this change. Right now this fixes a problem that's blocking regression testing.

Reply:
Ok, I created a PR https://github.com/silverstripe/silverstripe-framework/pull/10696 using the null safe operator. It's actually a handy operator since we can now rely on PHP8.1 for CMS5. I'd like to see it used more, will keep an eye out for opportunities.